api_server/modules/providers/slack_events_request.js

```
@@ -0,0 +1,279 @@
+'use strict';
+
+const crypto = require('crypto');
+const SlackUserHelper = require('./slack_user_helper');
+const RestfulRequest = require(process.env.CSSVC_BACKEND_ROOT + '/api_server/lib/util/restful/restful_request');
+const PostCreator = require(process.env.CSSVC_BACKEND_ROOT + '/api_server/modules/posts/post_creator');
+const PostDeleter = require(process.env.CSSVC_BACKEND_ROOT + '/api_server/modules/posts/post_deleter');
+const PostUpdater = require(process.env.CSSVC_BACKEND_ROOT + '/api_server/modules/posts/post_updater');
+const PostPublisher = require(process.env.CSSVC_BACKEND_ROOT + '/api_server/modules/posts/post_publisher');
+const PostIndexes = require(process.env.CSSVC_BACKEND_ROOT + '/api_server/modules/posts/indexes');
+
+class SlackEventsRequest extends RestfulRequest {
+
+	async authorize () {
+		if (!this.verifySlackRequest(this.request)) {
+			this.log('Slack verification failed');
+			throw this.errorHandler.error('notFound');
+		}
```

let's throw an `invalidParameter` error here instead, and change `log` to `warn`